src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v2/YFlowControllerV2.java

```diff
@@ -151,4 +163,79 @@ public CompletableFuture<YFlowPingResult> pingYFlow(
+        Stream<Optional<String>> checks = Stream.empty();
+        int idx = 0;
+        for (SubFlowUpdatePayload entry : subFlows) {
+            checks = Stream.concat(checks, verifySubFlowVlanIds(idx++, entry));
+        }
+        return checks;
+    }
+
+    private Stream<Optional<String>> verifyPatchSubFlowVlanIds(List<SubFlowPatchPayload> subFlows) {
+        Stream<Optional<String>> checks = Stream.empty();
+        int idx = 0;
+        for (SubFlowPatchPayload entry : subFlows) {
+            checks = Stream.concat(checks, verifySubFlowVlanIds(idx++, entry));
+        }
+        return checks;
+    }
+
+    private Stream<Optional<String>> verifySubFlowVlanIds(int idx, SubFlowUpdatePayload subFlow) {
+        String subId = formatSubFlowId(idx, subFlow.getFlowId());
+        YFlowSharedEndpointEncapsulation sharedEndpoint = subFlow.getSharedEndpoint();
+        String sharedEndpointReference = subId + "shared";
+        return Stream.concat(
+                verifyFlowEndpoint(
+                        subFlow.getEndpoint(), subId + "sub-endpoint"),
+                Stream.of(
+                        verifyEndpointVlanId(
+                                sharedEndpointReference, "vlanId", sharedEndpoint.getVlanId()),
+                        verifyEndpointVlanId(
+                                sharedEndpointReference, "innerVlanId", sharedEndpoint.getInnerVlanId())));
+    }
+
+    private Stream<Optional<String>> verifySubFlowVlanIds(int idx, SubFlowPatchPayload subFlow) {
+        String subId = formatSubFlowId(idx, subFlow.getFlowId());
+
+        Stream<Optional<String>> endpointChecks = Stream.empty();
+        FlowPatchEndpoint endpoint = subFlow.getEndpoint();
+        if (endpoint != null) {
+            String endpointReference = subId + "sub-endpoint";
+            endpointChecks = Stream.of(
+                    verifyOptionalEndpointVlanId(endpointReference, "vlanId", endpoint.getVlanId()),
+                    verifyOptionalEndpointVlanId(endpointReference, "innerVlanId", endpoint.getInnerVlanId()));
+        }
+
+        Stream<Optional<String>> sharedChecks = Stream.empty();
+        YFlowPatchSharedEndpointEncapsulation sharedEndpoint = subFlow.getSharedEndpoint();
+        if (sharedEndpoint != null) {
+            String endpointReference = subId + "shared";
+            sharedChecks = Stream.of(
+                    verifyOptionalEndpointVlanId(endpointReference, "vlanId", sharedEndpoint.getVlanId()),
+                    verifyOptionalEndpointVlanId(endpointReference, "innerVlanId", sharedEndpoint.getInnerVlanId()));
+        }
+
+        return Stream.concat(endpointChecks, sharedChecks);
+    }
+
+    private Optional<String> verifyOptionalEndpointVlanId(String endpoint, String field, Integer value) {
+        if (value != null) {
+            return verifyEndpointVlanId(endpoint, field, value);
+        }
+        return Optional.empty();
+    }
+
+    private String formatSubFlowId(int idx, String flowId) {
```

Comment:
If we have class that represents Flow/SubFlow, this method should be moved to the one.

Reply:
No. Because this is part of error reporting, this is the way how exactly this conftoller(rest API) reports error. It should not be part of generic "application" code.

Reply:
Do I get you right? According to your logic, `toString()` method should not be in Object class, but should be implemented in classes which need string representation, e.g. in loggers or the ones which output to the console?

Reply:
moved.